crates/sui-adapter/src/adapter.rs

```diff
@@ -336,28 +341,39 @@ pub fn publish<
         + ChildObjectResolver,
 >(
     state_view: &mut S,
-    vm: &MoveVM,
+    _vm: &MoveVM,
     natives: NativeFunctionTable,
     module_bytes: Vec<Vec<u8>>,
     ctx: &mut TxContext,
     gas_status: &mut GasStatus,
 ) -> Result<(), ExecutionError> {
-    let mut modules = module_bytes
+    let mut modules = make_compiled_modules(&module_bytes)?;
+    let package_id = generate_package_id(&mut modules, ctx)?;
+    let vm = verify_and_link(state_view, &modules, package_id, natives, gas_status)?;
+    store_package_and_init_modules(state_view, &vm, modules, ctx, gas_status)
+}
+
+pub fn make_compiled_modules(
+    module_bytes: &[Vec<u8>],
+) -> Result<Vec<CompiledModule>, ExecutionError> {
+    let modules = module_bytes
```

Comment:
I'm not sure I understand these changes. We could just deserialize over in the `transaction_input_checker`?